zone/corpse.cpp

```diff
@@ -2456,6 +2456,16 @@ void Corpse::ProcessLootLockouts(Client* give_exp_client, NPC* in_npc)
 						}
 						//if they're not in zone, this will be loaded once they are.
 						database.SaveCharacterLootLockout(playerItr->second.character_id, lootLockout.expirydate, in_npc->GetNPCTypeID(), in_npc->GetCleanName());
+						database.SaveCharacterInstanceLockout(playerItr->second.character_id, lootLockout.expirydate, zone->GetZoneID(), zone->GetGuildID());	
+
+						auto pack2 = new ServerPacket(ServerOP_CZUpdateLockoutCache, sizeof(CZUpdateLockoutCache_Struct));
+						CZUpdateLockoutCache_Struct* CZLC = (CZUpdateLockoutCache_Struct*)pack2->pBuffer;
+						CZLC->CharId = playerItr->second.character_id;
+						strn0cpy(CZLC->CharName, playerItr->second.character_name, 64);
+						CZLC->ZoneId = zone->GetZoneID();
+						CZLC->ZoneInstanceId = zone->GetGuildID();
+						worldserver.SendPacket(pack2);
+						safe_delete(pack2);
```

Comment:
This code is duplicated in a few places, so I wonder if it'd be better to make it its own function in case tweaks are needed in the future?

Reply:
Good call. It's in its own function now.